app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/appmanager/AppManagerActivity.java

```diff
@@ -213,8 +215,21 @@ public void onActivityResult(int requestCode, int resultCode, Intent resultData)
         if (requestCode == READ_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
             Intent startIntent = new Intent(AppManagerActivity.this, FwAppInstallerActivity.class);
             startIntent.setAction(Intent.ACTION_VIEW);
-            startIntent.setDataAndType(resultData.getData(), null);
-            startActivity(startIntent);
+            ClipData clipData = resultData.getClipData();
+            if (clipData != null) {
+                //If there is more than one app to install
+                int appsCount = clipData.getItemCount();
+                for (int i = 0; i < appsCount; i++) {
+                    startIntent.setDataAndType(clipData.getItemAt(i).getUri(), null);
+                    startIntent.putExtra("APPS_COUNT", appsCount);
+                    startIntent.putExtra("APP_INDEX", Math.abs(i - appsCount));
```

Comment:
Because we want it to be (1/3) -> (2/3) -> (3/3), not (3/3) -> (2/3) -> (1/3)
Why not use ```for (int i = appsCount; i >=0 ; i--)``` you ask?
That would be a good question.